libres/tests/enkf/test_cases_config.cpp

```diff
@@ -0,0 +1,40 @@
+#include "catch2/catch.hpp"
+
+#include <ert/enkf/cases_config.hpp>
+
+#include "../tmpdir.hpp"
+
+TEST_CASE("cases_config_get_and_set", "[enkf]") {
+    cases_config_type *cases_config = cases_config_alloc();
+
+    SECTION("cases_config_get_iteration_number") {
+        REQUIRE(cases_config_get_iteration_number(cases_config) == 0);
+        cases_config_free(cases_config);
+    }
+
+    SECTION("cases_config_set_int") {
+        cases_config_set_int(cases_config, "iteration_number", 12);
+        REQUIRE(cases_config_get_iteration_number(cases_config) == 12);
+        cases_config_free(cases_config);
+    }
+}
+
+bool Tmpdir::delete_temporary_files = true;
+
+TEST_CASE_METHOD(Tmpdir, "case_config_write_and_read", "[enkf]") {
+    GIVEN("cases_config") {
+        cases_config_type *cases_config = cases_config_alloc();
+
+        THEN("Write and read") {
```

Comment:
This was a very helpful example, thanks!

I've updated the code to use this and the new tmpdir.

Think I'll extend this PR to test all functions using `util_mkdir_fopen` so that I can later safely replace it with something more standard.